samples/drivers/video/tcpserversink/src/main.c

```diff
@@ -33,12 +33,171 @@ static ssize_t sendall(int sock, const void *buf, size_t len)
+		LOG_INF("  %s width [%u; %u; %u] height [%u; %u; %u]",
+			VIDEO_FOURCC_TO_STR(fcap->pixelformat), fcap->width_min, fcap->width_max,
+			fcap->width_step, fcap->height_min, fcap->height_max, fcap->height_step);
+		i++;
+	}
+
+	caps.type = VIDEO_BUF_TYPE_INPUT;
+	if (video_get_caps(encoder_dev, &caps)) {
+		LOG_ERR("Unable to retrieve video capabilities");
+		return -1;
+	}
+
+	LOG_INF("- Encoder input capabilities:");
+	i = 0;
+	while (caps.format_caps[i].pixelformat) {
+		const struct video_format_cap *fcap = &caps.format_caps[i];
+		/* fourcc to string */
+		LOG_INF("  %s width [%u; %u; %u] height [%u; %u; %u]",
+			VIDEO_FOURCC_TO_STR(fcap->pixelformat), fcap->width_min, fcap->width_max,
+			fcap->width_step, fcap->height_min, fcap->height_max, fcap->height_step);
+		i++;
+	}
+
+	/* Get default/native format */
+	fmt.type = VIDEO_BUF_TYPE_OUTPUT;
+	if (video_get_format(encoder_dev, &fmt)) {
+		LOG_ERR("Unable to retrieve video format");
+		return -1;
+	}
+
+	LOG_INF("Video encoder device detected, format: %s %ux%u",
+		VIDEO_FOURCC_TO_STR(fmt.pixelformat), fmt.width, fmt.height);
+
+#if CONFIG_VIDEO_FRAME_HEIGHT
+	fmt.height = CONFIG_VIDEO_FRAME_HEIGHT;
+#endif
+
+#if CONFIG_VIDEO_FRAME_WIDTH
+	fmt.width = CONFIG_VIDEO_FRAME_WIDTH;
+#endif
+
+	/* Set output format */
+	if (strcmp(CONFIG_VIDEO_ENCODED_PIXEL_FORMAT, "")) {
+		fmt.pixelformat = VIDEO_FOURCC_FROM_STR(CONFIG_VIDEO_ENCODED_PIXEL_FORMAT);
+	}
+
+	LOG_INF("- Video encoder output format: %s %ux%u", VIDEO_FOURCC_TO_STR(fmt.pixelformat),
+		fmt.width, fmt.height);
+
+	fmt.type = VIDEO_BUF_TYPE_OUTPUT;
+	if (video_set_format(encoder_dev, &fmt)) {
+		LOG_ERR("Unable to set format");
+		return -1;
+	}
+
+	/* Alloc output buffer */
+	size = fmt.size;
+	if (size == 0) {
+		LOG_ERR("Encoder driver must set format size");
+		return -1;
+	}
+
+	buffer = video_buffer_aligned_alloc(size, CONFIG_VIDEO_BUFFER_POOL_ALIGN, K_FOREVER);
+	if (buffer == NULL) {
+		LOG_ERR("Unable to alloc compressed video buffer size=%d", size);
+		return -1;
+	}
+
+	buffer->type = VIDEO_BUF_TYPE_OUTPUT;
+	video_enqueue(encoder_dev, buffer);
+
+	/* Set input format */
+	if (strcmp(CONFIG_VIDEO_PIXEL_FORMAT, "")) {
+		fmt.pixelformat = VIDEO_FOURCC_FROM_STR(CONFIG_VIDEO_PIXEL_FORMAT);
+	}
+
+	LOG_INF("- Video encoder input format: %s %ux%u", VIDEO_FOURCC_TO_STR(fmt.pixelformat),
+		fmt.width, fmt.height);
+
+	fmt.type = VIDEO_BUF_TYPE_INPUT;
+	if (video_set_format(encoder_dev, &fmt)) {
+		LOG_ERR("Unable to set input format");
+		return -1;
+	}
+
+	/* Start video encoder */
+	if (video_stream_start(encoder_dev, VIDEO_BUF_TYPE_INPUT)) {
+		LOG_ERR("Unable to start video encoder (input)");
+		return -1;
+	}
+	if (video_stream_start(encoder_dev, VIDEO_BUF_TYPE_OUTPUT)) {
+		LOG_ERR("Unable to start video encoder (output)");
+		return -1;
+	}
+
+	return 0;
+}
+
+int encode_frame(struct video_buffer *in, struct video_buffer **out)
+{
+	int ret;
+
+	in->type = VIDEO_BUF_TYPE_INPUT;
+	video_enqueue(encoder_dev, in);
```

Comment:
It looks like STM32N6 VENC [can fail](https://github.com/zephyrproject-rtos/zephyr/pull/92884/files#diff-58fac910c9b0c6b671578694deba98095425bc473dc36fa6f49e61ec91ea9db3R675). Worth checking the error code here too? Or should it be ignored on purpose?